### PR TITLE
[LTC] Upstream several internal ops

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69815
* __->__ #69716
* #69621

Summary: To prepare for the landing of LazyTensor and LazyGraphExecutor,
- arithmetic_ir_ops.h
- cast.h
- device_data.h
- expand.h
- generic.h
- scalar.h

Differential Revision: [D32999410](https://our.internmc.facebook.com/intern/diff/D32999410)

### CODE REVIEW

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32999410).

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32999410).

File: torch/csrc/lazy/core/internal_ops/arithmetic_ir_ops.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#include <torch/csrc/lazy/core/helpers.h>
+
+#include <memory>
+
+#include <torch/csrc/lazy/core/internal_ops/generic.h>
+
+namespace torch {
+namespace lazy {
+
+NodePtr operator+(const Value& node1, const Value& node2) {
```
Comment (wconstab) [inline torch/csrc/lazy/core/internal_ops/arithmetic_ir_ops.cpp:12]: these ops in particular are ones i think we don't want to put in core.  keeping them in ts_backend folder for is fine, but hopefully we'll be able to delete them soon once we codegen mul and a few other ops.

**Review by wconstab — APPROVED** [review]
would prefer to locate the ops under ts_backend but otherwise lgtm.

**Comment by desertfire** [comment]
> would prefer to locate the ops under ts_backend but otherwise lgtm.

Moved based on the suggestion.

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32999410).

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32999410).

**Pull request closed without merging**